src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h

```diff
@@ -0,0 +1,165 @@
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/FreeFloatingState.h>
+#include <iDynTree/Model/Model.h>
+
+// Eigen
+#include <Eigen/Dense>
+
+// RDE
+#include <BipedalLocomotion/RobotDynamicsEstimator/SubModel.h>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+namespace RobotDynamicsEstimator
+{
+
+/**
+ * SubModelKinDynWrapper is a concrete class and implements a wrapper of the KinDynComputation class
+ * from iDynTree. The class is used to take updated the sub-model kinematics and dynamics
+ */
+
+class SubModelKinDynWrapper
+{
+
+private:
+    SubModel subModel; /**< SubModel struct containing all the information about the sub-model */
+    iDynTree::KinDynComputations kinDyn; /**< kindyncomputation object used to get the sub-model
+                                            kinematic and dynamic information */
+    iDynTree::MatrixDynSize massMatrix; /**< Mass matrix of the sub-model */
+    iDynTree::FreeFloatingGeneralizedTorques genForces; /**< Generalized force vector */
+    std::map<std::string, iDynTree::MatrixDynSize> jacFTList; /**< Jacobians of the FT sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacAccList; /**< Jacobians of the accelerometer
+                                                                  sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacGyroList; /**< Jacobians of the gyroscope
+                                                                   sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacContactList; /**< Jacobians of the external
+                                                                      contacts */
+    std::map<std::string, iDynTree::Vector6> dJnuList; /**< Accelerometer bias accelerations */
+    std::map<std::string, iDynTree::Matrix3x3> accRworldList; /**< Rotation matrix of the
+                                                                 accelerometer frame wrt world */
+
+protected:
+    /**
+     * updateDynamicsVariableState updates the value of all the member variables containing
+     * information about the robot kinematics and dynamics
+     */
+    bool updateDynamicsVariableState(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+
+public:
+    /**
+     * @brief initialize initializes the kindyncomputation object and resizes
+     * all the variable not allocated yet.
+     * @param subModel a struct of type SubModif el describing the subModel structure and model.
+     * @return a boolean value saying if all the variables are initialized correctly.
+     */
+    bool initialize(SubModel subModel);
+
+    /**
+     * @brief updateKinDynState updates the state of the KinDynWrapper object.
+     * @param kinDynFullModel is the KinDynComputation object associated to a model.
+     * @return a boolean value saying if the subModelList has been created correctly.
+     */
+    bool updateKinDynState(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
```

Comment:
I guess that we are not expecting ``kinDynFullModel`` to be modified by this method. If so, I would suggest changing the input to ``std::shared_ptr<const iDynTree::KinDynComputations>``

Reply:
If I use `std::shared_ptr<const iDynTree::KinDynComputations>` for `kinDynFullModel` I have this error:

```bash
error: passing ‘std::__shared_ptr_access<const iDynTree::KinDynComputations, __gnu_cxx::_S_atomic, false, false>::element_type’ {aka ‘const iDynTree::KinDynComputations’} as ‘this’ argument discards qualifiers [-fpermissive]
   88 |     iDynTree::Transform worldTBase = kinDynFullModel->getWorldTransform(baseFrameName);

```

Reply:
Ah dammit. This is because that method is not marked as ``const``. See https://github.com/robotology/idyntree/blob/85fa0cfedbb0e3029b194c7afd3bf906242ea3c6/src/high-level/include/iDynTree/KinDynComputations.h#L467

That's because internally it can trigger some computation. I guess this goes beyond the scope of this PR. Feel free to ignore the original comment :wink: